src/lib/trust_system/peer_trust.ml

```diff
@@ -122,18 +132,16 @@ struct
                   ) ]
               @ action_metadata )
             "Banning peer $peer until $expiration because it %s" action_fmt ;
-          if Option.is_some db then Strict_pipe.Writer.write bans_writer peer
+          if Option.is_some db then (
+            Coda_metrics.Gauge.inc_one Coda_metrics.Trust_system.banned_peers ;
+            Strict_pipe.Writer.write bans_writer peer )
           else Deferred.unit
-      | _, _ ->
-          let verb =
-            if simple_new.trust >. simple_old.trust then "Increasing"
-            else "Decreasing"
-          in
-          Logger.debug logger ~module_:__MODULE__ ~location:__LOC__
-            ~metadata:([("peer", Peer_id.to_yojson peer)] @ action_metadata)
-            "%s trust for peer $peer due to action %s. New trust is %f." verb
-            action_fmt simple_new.trust ;
+      | Banned_until _, Unbanned ->
+          Coda_metrics.Gauge.dec_one Coda_metrics.Trust_system.banned_peers ;
```

Comment:
This function is only called when we check if a peer is banned, so the metric is going to substantially overcount. If we want the metric to be accurate we need to either scan the db at regular intervals on a timer or put something like `don't_wait_for (Async.after 24hrs decrement_metric)` after the banning code.